File: JDK11/opentraceability/src/main/java/opentraceability/models/events/kdes/eventkdeboolean.java
package opentraceability.models.events.kdes;

import opentraceability.Constants;
import opentraceability.interfaces.IEventKDE;
import org.json.JSONObject;
import org.w3c.dom.Element;

import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import java.lang.reflect.Type;

public class EventKDEBoolean implements IEventKDE {

    public Type valueType = Boolean.class;

    public Boolean value;

    public EventKDEBoolean() {
    }

    public EventKDEBoolean(String ns, String name) {
        this.namespace = ns;
        this.name = name;
    }

    @Override
    public Object getJson() {
        if (value != null) {
            return new JSONObject().put("value", value);
        } else {
            return null;
        }
    }

    @Override
    public Element getXml() throws ParserConfigurationException {
        if (value != null) {
            var documentBuilderFactory = DocumentBuilderFactory.newInstance();
            var documentBuilder = documentBuilderFactory.newDocumentBuilder();
            var document = documentBuilder.newDocument();
            var element = document.createElementNS(namespace, name);
            element.setAttributeNS(Constants.XSI_NAMESPACE, "type", "boolean");
            element.setTextContent(value.toString());
            return element;
        } else {
            return null;
        }
    }

    @Override
    public void setFromJson(JSONObject json) {
        if (json.has("value")) {
            this.value = json.getBoolean("value");
        }
    }

    @Override
    public void setFromXml(Element xml) {
        if (xml != null) {
            this.value = Boolean.parseBoolean(xml.getTextContent());
        }
    }

    @Override
    public String toString() {
        return value != null ? value.toString() : "";
    }
}

File: Kotlin/OpenTraceability/src/main/kotlin/opentraceability/models/events/kdes/eventkdeboolean.kt
package opentraceability.models.events.kdes

import opentraceability.interfaces.IEventKDE
import opentraceability.Constants
import org.json.JSONObject
import org.w3c.dom.Element
import java.lang.reflect.Type
import javax.xml.parsers.DocumentBuilderFactory
import kotlin.reflect.KType
import kotlin.reflect.full.starProjectedType

class EventKDEBoolean() : EventKDEBase(), IEventKDE {

    override var valueType: KType = Boolean::class.starProjectedType

    var value: Boolean? = null


    constructor(ns: String, name: String) : this() {
        this.namespace = ns
        this.name = name
    }

    override fun getJson(): Any? {
        value?.let { return JSONObject().put("value", it) }
        return null
    }

    override fun getXml(): Element? {
        value?.let {
            val documentBuilderFactory = DocumentBuilderFactory.newInstance()
            val documentBuilder = documentBuilderFactory.newDocumentBuilder()
            val document = documentBuilder.newDocument()
            val element = document.createElementNS(namespace, name)
            element.setAttributeNS(Constants.XSI_NAMESPACE, "type", "boolean")
            element.textContent = it.toString()
            return element
        }
        return null
    }

    override fun setFromJson(json: JSONObject) {
        this.value = json["value"] as Boolean
    }

    override fun setFromXml(xml: Element) {
        this.value = xml.textContent.toBoolean()
    }

    override fun toString(): String {
        return value?.toString() ?: ""
    }

}
